### PR TITLE
Tasks 2.6 - 2.7

### PR DESCRIPTION
**Pull request by mikeivashka**

### CODE REVIEW

File: core/src/main/java/com/es/core/cart/HttpSessionCartService.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public void addPhone(Long phoneId, Long quantity) {
 
     @Override
     public void update(Map<Long, Long> items) {
-        Set<CartItem> cartItemsToBeSet = new HashSet<>(items.size());
+        Set<CartItem> cartItemsToBeSet = new LinkedHashSet<>(items.size());
```
Comment (marty-friedman) [inline core/src/main/java/com/es/core/cart/HttpSessionCartService.java]: Please use Stream#collect

Comment (marty-friedman) [reply core/src/main/java/com/es/core/cart/HttpSessionCartService.java]: \+ Not in scope of this task, but rather a previous one:
there is no point in retrieving the product from DAO in the #remove method of this class

File: web/src/main/java/com/es/phoneshop/web/controller/pages/CartPageController.java [hunk 2]
```diff
@@ -16,21 +21,44 @@
...
-        cartService.getCart();
+    public String getCart(Model model) {
+        Map<Phone, Long> cart = cartService.getCart();
+        model.addAttribute("cart", cart);
+        return "cartPage";
     }
 
-    @GetMapping(value = "/minicart")
+    @PutMapping
+    public String updateCart(@ModelAttribute CartUpdateForm form, HttpServletRequest req, Model model) {
```
Comment (marty-friedman) [inline web/src/main/java/com/es/phoneshop/web/controller/pages/CartPageController.java]: Please use spring MVC validation to validate quantities according to the task requirements

File: core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HttpSessionCartServiceUnitTest {
```
Comment (marty-friedman) [inline core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java:21]: This test class only covers two recalculation methods of the service, please provide 100% code coverage

File: core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    @InjectMocks
+    @Spy
+    private HttpSessionCartService cartService;
+
+    @Mock
+    private Cart cart;
+
+    @Test
+    public void getCartTotalPriceTest() {
+        double[] prices = {200, 350, 400.5f, 150};
```
Comment (marty-friedman) [inline core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java]: Don't do that in tests. Ideally, all mocks must be defined as `@Mock` fields of the test class (phone1, phone2, ... or whatever) and set up in the `@Before` method according to the happy path or more or less the common state. The test method should normally only consist of 2 or 3 extremely short parts:
1) (optional) Change the default mock setup defined in the `@Before` method, to test some boundary condition (not-happy-path tests)
2) Invoke the tested method. If applicable, save its result to a variable
3) Verify the result using assert* / verify methods

Comment (marty-friedman) [reply core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java]: So, in fact, after all changes this method should look like:
```java
when(cart.getCartItems()).thenReturn(Arrays.asList(phone1, phone2, phone3, phone4)); // if not set up in @Before, which better to be, actually

BigDecimal result = cartService.getTotalPrice();

assertEquals(TOTAL_PRICE, result); // where TOTAL_PRICE is a class constant
```

File: core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        }
+        BigDecimal expected = sum;
+        BigDecimal actual = cartService.getTotalPrice();
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void getCartTotalItemsCountTest() {
+        long[] quantities = {2, 2, 1, 3};
```
Comment (marty-friedman) [inline core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java]: Same thing here

File: core/src/main/java/com/es/core/cart/HttpSessionCartService.java (hunk 1 above)
Comment (mikeivashka) [reply core/src/main/java/com/es/core/cart/HttpSessionCartService.java]: Fixed

File: core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java (hunk 4 above)
Comment (mikeivashka) [reply core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java]: fixed

File: core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java (hunk 5 above)
Comment (mikeivashka) [reply core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java]: fixed

File: core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java (hunk 3 above)
Comment (mikeivashka) [reply core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java:21]: fixed
